openssl/src/hash.rs

```diff
@@ -106,6 +106,11 @@ impl MessageDigest {
     pub fn size(&self) -> usize {
         unsafe { ffi::EVP_MD_size(self.0) as usize }
     }
+
+    /// The name of the digest
+    pub fn nid(&self) -> Nid {
```

Comment:
Can we call this `type_` to match with the openssl function name?

Reply:
OpenSSL is generally inconsistent on their "give me the NID for this type" functions. I'd prefer that rust-openssl fixes this problem, especially given that their function names collide with Rust keywords.

Reply:
Our general policy is to follow the openssl names pretty directly.